api/engine.py

```diff
@@ -322,6 +330,79 @@ def __get_result_content():
                                    msg="your scan id is not valid!")), 400
     return __get_result(__language(), id)
 
+@app.route("/results/get_json", methods=["GET"])
+def __get_results_json():
+    """
+    get host's logs through the API in JSON type
+
+    Returns:
+        an array with JSON events
+    """
+    session = create_connection(__language())
+    __api_key_check(app, flask_request, __language())
+    try:
+        id = int(__get_value(flask_request, "id"))
+        scan_id_temp = session.query(Report).filter(Report.id==id).all()
+    except Exception as _:
+        id = ""
+    if(scan_id_temp):
+        result_id = session.query(Report).join(HostsLog, Report.scan_id==HostsLog.scan_id).filter(Report.scan_id==scan_id_temp[0].scan_id).all()
+    else:
+        result_id = []
+    json_object = {}
+    if(result_id):
+        scan_id = result_id[0].scan_id
+        data = __logs_by_scan_id(scan_id, __language())
+        json_object = json.dumps(data)
+    filename = "report-" + datetime.now().strftime("%Y%m%d-%H%M%S")
+    return Response(json_object,
+        mimetype='application/json',
+        headers={'Content-Disposition':'attachment;filename='+filename+'.json'})
+
+@app.route("/results/get_csv", methods=["GET"])
+def __get_results_csv():
+    """
+    get host's logs through the API in JSON type
+
+    Returns:
+        an array with JSON events
+    """
+    session = create_connection(__language())
+    __api_key_check(app, flask_request, __language())
+    try:
+        id = int(__get_value(flask_request, "id"))
+        scan_id_temp = session.query(Report).filter(Report.id==id).all()
+    except Exception as _:
+        id = ""
+    if(scan_id_temp):
+        result_id = session.query(Report).join(HostsLog, Report.scan_id==HostsLog.scan_id).filter(Report.scan_id==scan_id_temp[0].scan_id).all()
+    else:
+        result_id = []
+    s = ''
+    if(result_id):
+        scan_id = result_id[0].scan_id
+        data = __logs_by_scan_id(scan_id, __language())
+        keys = data[0].keys()
+        with open('results.csv', "w")  as output_file:
```

Comment:
@aman566  filename here is still hardcoded results.csv - also please check where are you saving it? please use the standard used  in core/config.py


https://github.com/OWASP/Nettacker/blob/3bd9d11918c276093959efbe70005ab8aa0762dd/core/config.py#L118